mllib/src/main/scala/org/apache/spark/mllib/fpm/AssociationRules.scala

```diff
@@ -120,6 +120,13 @@ object AssociationRules {
     @Since("1.5.0")
     def confidence: Double = freqUnion.toDouble / freqAntecedent
 
+    /**
+     * Returns the support of the rule. Current implementation would return the number of
+     * co-occurrence of antecedent and consequent.
+     */
+    @Since("2.1.0")
+    def support: Double = freqUnion.toDouble
+
```

Comment:
Ah right, we use support as a fraction. Well, then best to be consistent and return it as a fraction of the data set size. I can't imagine having a method sometimes return a value with one type of semantics and sometimes another. Just make two methods.

freqUnion however appears to be a count only, and is even explicitly called a 'frequency'.